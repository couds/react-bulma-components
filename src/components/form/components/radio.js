import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import events from '../../../modifiers/events';

export default class Radio extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    style: PropTypes.object,
    /**
     * The name of the input field Commonly used for [multi-input handling](https://reactjs.org/docs/forms.html#handling-multiple-inputs)
     */
    name: PropTypes.string.isRequired,
    /**
     * True disable this input
     */
    disabled: PropTypes.bool,
    /**
     * True if this checkbox is checked
     */
    checked: PropTypes.bool,
    /**
     * The form value of this radio button
     */
    value: PropTypes.any,
    ...events.propTypes,
  }

  static defaultProps = {
    children: null,
    className: '',
    value: '',
    style: {},
    disabled: false,
    checked: false,
    ...events.defaultProps,
  }

  render() {
    const {
      className,
      style,
      disabled,
      checked,
      value,
      name,
      children,
      ...props
    } = this.props;
    return (
      <label
        disabled={disabled}
        className={classnames('radio', className)}
        style={style}
      >
        <input
          {...events.props(props)}
          name={name}
          checked={checked}
          type="radio"
          value={value}
          disabled={disabled}
        />
        {children}
      </label>
    );
  }
}
