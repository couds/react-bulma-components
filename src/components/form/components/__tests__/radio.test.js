import React from 'react';
import { render } from 'react-testing-library';
import { Radio } from '../radio';

describe('Radio component', () => {
  it('should render', () => {
    const { asFragment } = render(
      <Radio name="test">
        Test <a>Give me</a>
      </Radio>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should concat classname in props with classname', () => {
    const { asFragment } = render(
      <Radio name="test" className="other-class this-is-a-test">
        <p>Default</p>
      </Radio>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should use inline styles', () => {
    const { asFragment } = render(
      <Radio name="test" style={{ height: 250 }}>
        <p>Default</p>
      </Radio>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should be disabled, checked and with value', () => {
    const { asFragment } = render(
      <Radio name="test" value="TEST" checked disabled>
        <p>Default</p>
      </Radio>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
