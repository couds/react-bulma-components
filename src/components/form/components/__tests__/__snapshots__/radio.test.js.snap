// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio component should be disabled, checked and with value 1`] = `
<DocumentFragment>
  <label
    class="b-radio radio"
    disabled=""
  >
    <input
      checked=""
      disabled=""
      name="test"
      type="radio"
      value="TEST"
    />
    <span
      class="check"
    />
    <span
      class="control-label"
    >
      <p>
        Default
      </p>
    </span>
  </label>
</DocumentFragment>
`;

exports[`Radio component should concat classname in props with classname 1`] = `
<DocumentFragment>
  <label
    class="b-radio radio other-class this-is-a-test"
  >
    <input
      name="test"
      type="radio"
      value=""
    />
    <span
      class="check"
    />
    <span
      class="control-label"
    >
      <p>
        Default
      </p>
    </span>
  </label>
</DocumentFragment>
`;

exports[`Radio component should render 1`] = `
<DocumentFragment>
  <label
    class="b-radio radio"
  >
    <input
      name="test"
      type="radio"
      value=""
    />
    <span
      class="check"
    />
    <span
      class="control-label"
    >
      Test 
      <a>
        Give me
      </a>
    </span>
  </label>
</DocumentFragment>
`;

exports[`Radio component should use inline styles 1`] = `
<DocumentFragment>
  <label
    class="b-radio radio"
    style="height: 250px;"
  >
    <input
      name="test"
      type="radio"
      value=""
    />
    <span
      class="check"
    />
    <span
      class="control-label"
    >
      <p>
        Default
      </p>
    </span>
  </label>
</DocumentFragment>
`;
